src/compiler/types.ts

```diff
@@ -4173,6 +4173,18 @@ export interface SourceFileLike {
     getPositionOfLineAndCharacter?(line: number, character: number, allowEdits?: true): number;
 }
 
+/** @internal */
+export interface FutureSourceFile {
+    readonly path: Path;
+    readonly fileName: string;

```

Comment:
Does this need to have resolvedPath and originalFileName as well ? are they used anywhere.. it would be needed if this is from other project that is referenced and i am not sure if thats something is needed as future file or not.

Reply:
I added only fields that were necessary to add for the current usage, with the expectation that other optional/empty fields be added in the future if needed.